src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -1751,33 +1679,73 @@ subroutine initialize_ice_shelf(param_file, ocn_grid, Time, CS, diag, forces_in,
   call pass_var(G%bathyT, G%domain)
   call cpu_clock_end(id_clock_pass)
 
-
   do j=jsd,jed ; do i=isd,ied
     if (ISS%area_shelf_h(i,j) > G%areaT(i,j)) then
       call MOM_error(WARNING,"Initialize_ice_shelf: area_shelf_h exceeds G%areaT.")
       ISS%area_shelf_h(i,j) = G%areaT(i,j)
     endif
   enddo ; enddo
-  if (present(fluxes_in)) then ; do j=jsd,jed ; do i=isd,ied
-    if (G%areaT(i,j)>0.) fluxes%frac_shelf_h(i,j) = ISS%area_shelf_h(i,j) / G%areaT(i,j)
-  enddo ; enddo ; endif
 
   if (CS%debug) then
-    call hchksum(fluxes%frac_shelf_h, "IS init: frac_shelf_h", G%HI, haloshift=0)
     call hchksum(ISS%area_shelf_h, "IS init: area_shelf_h", G%HI, haloshift=0, scale=US%L_to_m*US%L_to_m)
   endif
 
-  if (present(forces_in)) &
-    call add_shelf_forces(G, US, CS, forces, do_shelf_area=.not.CS%solo_ice_sheet)
 
-  if (present(fluxes_in)) call add_shelf_pressure(ocn_grid, US, CS, fluxes)
+  ! Set up the restarts.
+
+  call restart_init(param_file, CS%restart_CSp, "Shelf.res")
+  call register_restart_field(ISS%mass_shelf, "shelf_mass", .true., CS%restart_CSp, &
+                              "Ice shelf mass", "kg m-2")
+  call register_restart_field(ISS%area_shelf_h, "shelf_area", .true., CS%restart_CSp, &
+                              "Ice shelf area in cell", "m2")
+  call register_restart_field(ISS%h_shelf, "h_shelf", .true., CS%restart_CSp, &
+                              "ice sheet/shelf thickness", "m")
+  if (PRESENT(sfc_state_in)) then
+    if (allocated(sfc_state%taux_shelf) .and. allocated(sfc_state%tauy_shelf)) then
+       u_desc = var_desc("taux_shelf", "Pa", "the zonal stress on the ocean under ice shelves", &
+            hor_grid='Cu',z_grid='1')
+       v_desc = var_desc("tauy_shelf", "Pa", "the meridional stress on the ocean under ice shelves", &
+            hor_grid='Cv',z_grid='1')
+       call register_restart_pair(sfc_state%taux_shelf, sfc_state%tauy_shelf, u_desc,v_desc, &
+            .false., CS%restart_CSp)
+    endif
+  endif
+
+  call register_restart_field(ISS%h_shelf, "_shelf", .true., CS%restart_CSp, &
```

Comment:
This field is being registered twice. It was already registered above as "h_shelf".

Reply:
Thank you very much for catching this duplicate call. This PR has been updated to remove the duplicate call.

  We had a significant breach in our testing protocols when we accepted PR #1265, which allowed this problem and the others that are addressed by this PR to slip into the dev/gfdl version of the MOM6 code.  The problems with that PR included widespread structural changes mixed in with answer changes with no run-time option to recover previous answers, limited discussion and no group consensus on significant restructuring, forced changes to top-level interfaces without adequate coordination across the various institutional forks, a lack of enforcement of MOM6 code style standards, and enormous commits within the PR that did not facilitate debugging. I hope that we have now corrected most of the fallout from PR #1265, but it has clearly exposed weaknesses in our testing and evaluation procedures that we will need to correct.